tendermint/src/chain/id.rs

```diff
@@ -2,89 +2,100 @@
 
 use crate::error::{Error, Kind};
 use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+use std::convert::TryFrom;
 use std::{
     cmp::Ordering,
     fmt::{self, Debug, Display},
     hash::{Hash, Hasher},
     str::{self, FromStr},
 };
+use tendermint_proto::DomainType;
 
 /// Maximum length of a `chain::Id` name. Matches `MaxChainIDLen` from:
 /// <https://github.com/tendermint/tendermint/blob/develop/types/genesis.go>
 // TODO: update this when `chain::Id` is derived from a digest output
 pub const MAX_LENGTH: usize = 50;
 
 /// Chain identifier (e.g. 'gaia-9000')
-#[derive(Copy, Clone)]
-pub struct Id([u8; MAX_LENGTH]);
```

Comment:
Recording it for any outsiders:
We discussed this on Slack and the reason might be because of the `Copy` trait (which was removed when this became a String). The `Copy` trait is not available for Strings. But Strings are more convenient to use in the code and the Clone trait is good enough for copying the structure where necessary.